### DISCUSSION TITLE
How to manipulate surrounding symbols?

### DISCUSSION THREAD

**Discussion by jdnavarro** [meow-edit/meow#205] [Q&A] [answered]
Coming from [Helix](https://helix-editor.com/) , I'm missing something like [its surround functionality](https://docs.helix-editor.com/usage.html#surround).

For surround symbols which are the same, like `"`,  I thought of using beacon mode, but how do you insert fake cursors at the beginning and end of a region?

**Comment by DogLooksGood** [comment, marked as answer]
No, there's not. But Emacs has a lot packages for manipulating parentheses, like `smartparens` or `paredit`.

For vanilla Emacs, there's a command called `insert-pair`.
```
(global-set-key (kbd "M-(") 'insert-pair)
(global-set-key (kbd "M-\"") 'insert-pair)
(global-set-key (kbd "M-{") 'insert-pair)
(global-set-key (kbd "M-[") 'insert-pair)
```
